sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -1348,4 +1349,37 @@ public static SentenceAssessment findSentimentAssessment(String assessmentPointe
         }
         return assessments.get(assessmentIndex);
     }
+
+    /**
+     * Throw exception if sourceVersion is one of target API versions.
+     *
+     * @param sourceVersion The service version that client is using when calling service API.
+     * @param targetVersions The target versions that used to verify if sourceVersion is unsupported version.
+     *                       Throw exception if target versions contains the source version.
+     * @param errorMessage The error message.
+     */
+    public static void throwIfTargetServiceVersionFound(TextAnalyticsServiceVersion sourceVersion,
+                                                        List<TextAnalyticsServiceVersion> targetVersions,
+                                                        String errorMessage) {
+        for (TextAnalyticsServiceVersion targetVersion : targetVersions) {
+            if (targetVersion != null && sourceVersion != null
+                && targetVersion.getVersion().equals(sourceVersion.getVersion())) {
+                throw LOGGER.logExceptionAsError(new IllegalStateException(errorMessage));
```

Comment:
Am I reading this correctly that it throws in every loop iteration? Shouldn't this function throw just once with an error message that has all the target service versions?

Reply:
Never mind, I just looked at how it is being used again and it will throw at max only once since the input service version can only match one of the input target versions.

Reply:
In java. the uncaught exception will terminate the thread. so this for loop will not continue to run. The method will exit immediately.